src/sidebar/components/SearchInput.tsx

```diff
@@ -1,103 +1,33 @@
-import {
-  IconButton,
-  Input,
-  SearchIcon,
-  Spinner,
-} from '@hypothesis/frontend-shared';
+import { Input, useSyncedRef } from '@hypothesis/frontend-shared';
 import classnames from 'classnames';
 import type { RefObject } from 'preact';
-import { useCallback, useRef, useState } from 'preact/hooks';
+import { useState } from 'preact/hooks';
 
-import { useShortcut } from '../../shared/shortcut';
-import { isMacOS } from '../../shared/user-agent';
 import { useSidebarStore } from '../store';
 
-/**
- * Respond to keydown events on the document (shortcut keys):
- *
- * - Focus the search input when the user presses '/', unless the user is
- *   currently typing in or focused on an input field.
- * - Focus the search input when the user presses CMD-K (MacOS) or CTRL-K
- *   (everyone else)
- * - Restore previous focus when the user presses 'Escape' while the search
- *   input is focused.
- */
-function useSearchKeyboardShortcuts(
-  searchInputRef: RefObject<HTMLInputElement>,
-) {
-  const prevFocusRef = useRef<HTMLOrSVGElement | null>(null);
-
-  const focusSearch = useCallback(
-    (event: KeyboardEvent) => {
-      // When user is in an input field, respond to CMD-/CTRL-K keypresses,
-      // but ignore '/' keypresses
-      if (
-        !event.metaKey &&
-        !event.ctrlKey &&
-        event.target instanceof HTMLElement &&
-        ['INPUT', 'TEXTAREA'].includes(event.target.tagName)
-      ) {
-        return;
-      }
-      prevFocusRef.current = document.activeElement as HTMLOrSVGElement | null;
-      if (searchInputRef.current) {
-        searchInputRef.current?.focus();
-        event.preventDefault();
-        event.stopPropagation();
-      }
-    },
-    [searchInputRef],
-  );
-
-  const restoreFocus = useCallback(() => {
-    if (document.activeElement === searchInputRef.current) {
-      if (prevFocusRef.current) {
-        prevFocusRef.current.focus();
-        prevFocusRef.current = null;
-      }
-      searchInputRef.current?.blur();
-    }
-  }, [searchInputRef]);
-
-  const modifierKey = isMacOS() ? 'meta' : 'ctrl';
-
-  useShortcut('/', focusSearch);
-  useShortcut(`${modifierKey}+k`, focusSearch);
-  useShortcut('escape', restoreFocus);
-}
-
 export type SearchInputProps = {
-  /**
-   * When true, the input field is always shown. If false, the input field is
-   * only shown if the query is non-empty.
-   */
-  alwaysExpanded?: boolean;
-
   /** The currently-active filter query */
   query: string | null;
 
   /** Callback for when the current filter query changes */
   onSearch: (value: string) => void;
+
+  /** The input's ref object, in case it needs to be handled by consumers */
+  inputRef?: RefObject<HTMLInputElement | undefined>;
```

Comment:
This allows the new `SearchPanel` component handle the input reference, and pass it to the `SidebarPanel`, making sure the input is focused on open.